GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -905,10 +905,27 @@ private async Task UpdateBranchNameDisplayAsync()
             await TaskScheduler.Default;
 
             var currentBranchName = Module.GetSelectedBranch();
+            var currentBranch = Module.GetRefs(false, true).FirstOrDefault(r => r.LocalName == currentBranchName);
+            string pushTo;
+            if (string.IsNullOrEmpty(currentBranch.TrackingRemote) || string.IsNullOrEmpty(currentBranch.MergeWith))
+            {
+                pushTo = new TranslationString(" -> (No remote branch configured)").Text;
+            }
+            else
+            {
+                var pushToTranslation = new TranslationString(" -> {0}/{1}");
+                pushTo = string.Format(pushToTranslation.Text,
+                                      currentBranch.TrackingRemote,
+                                      currentBranch.MergeWith);
+            }
 
             await this.SwitchToMainThreadAsync();
 
-            branchNameLabel.Text = currentBranchName;
+            branchNameLabel.Click += (object sender, EventArgs e) =>
+            {
+                UICommands.StartRemotesDialog(this, null, currentBranchName);
+            };
+            branchNameLabel.Text = currentBranchName + pushTo;
```

Comment:
This doesn't look right at all, I think you may have misinterpreted my feedback...
* `TranslationString` must be defined up the top of the class, please have a look how they are defined and used
* `pushTo` must be the name of the branch or the literal
* `branchNameLabel.Text = string.Format(...)` as per my earlier feedback